test/domain.middleware.test.ts

```diff
@@ -0,0 +1,55 @@
+import 'mocha';
+import { Request, Response, NextFunction } from 'express';
+import { expect } from 'chai';
+import { createRequest, createResponse } from 'node-mocks-http';
+
+import * as globalStore from '../src';
+import AsyncStoreAdapter from '../src/AsyncStoreAdapter';
+
+type Imiddleware = (req: Request, res: Response, next: NextFunction) => Promise<any> | void;
+
+describe('store: [adapter=DOMAIN] middleware', () => {
```

Comment:
Instead of creating a new file `test/domain.middleware.test.ts`, you can move all these tests under the same file `test/domain.test.ts`. We wanted to put all the tests for the domain based implementation to that file.

Later when we have a new adapter we'll make a new file.